src/IR.cpp

```diff
@@ -537,6 +537,10 @@ Expr Call::make(Function func, const std::vector<Expr> &args, int idx) {
 Expr Call::make(Type type, const std::string &name, const std::vector<Expr> &args, CallType call_type,
                 IntrusivePtr<FunctionContents> func, int value_index,
                 Buffer<> image, Parameter param) {
+    if (name == Call::prefetch) {
```

Comment:
Good point. Then (name == Call::Prefetch && call_type == Call::Intrinsic)

We don't have any pass in the compiler that prevents people from naming Funcs the same as our intrinsics.